includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -285,6 +287,56 @@ public function process_redirect_payment( $order_id, $intent_id, $save_payment_m
 		}
 	}
 
+	/**
+	 * Set formatted readable payment method title for order,
+	 * using payment method details from accompanying charge.
+	 *
+	 * @param WC_Order $order WC Order being processed.
+	 * @param array    $payment_method_details Array of payment method details from charge.
+	 */
+	public function set_payment_method_title_for_order( $order, $payment_method_details ) {
+		if ( empty( $payment_method_details ) ) {
+			return;
+		}
+
+		switch ( $payment_method_details['type'] ) {
+			case 'card':
+				$details       = $payment_method_details['card'];
+				$funding_types = [
+					'credit'  => __( 'credit', 'woocommerce-payments' ),
+					'debit'   => __( 'debit', 'woocommerce-payments' ),
+					'prepaid' => __( 'prepaid', 'woocommerce-payments' ),
+					'unknown' => __( 'unknown', 'woocommerce-payments' ),
+				];
+
+				$payment_method_title = sprintf(
+					// Translators: %1$s card brand, %2$s card funding (prepaid, credit, etc.).
+					__( '%1$s %2$s card', 'woocommerce-payments' ),
+					ucfirst( $details['network'] ),
+					$funding_types[ $details['funding'] ]
+				);
+				break;
+
+			case 'giropay':
+				$payment_method_title = 'Giropay';
+				break;
+
+			case 'sepa_debit':
+				$payment_method_title = 'SEPA Direct Debit';
+				break;
+
+			case 'sofort':
+				$payment_method_title = 'Sofort';
+				break;
+
+			default:
+				return;
+		}
+
+		$order->set_payment_method_title( "$payment_method_title (WooCommerce Payments)" );
```

Comment:
Adding the `(WooCommerce Payments)` suffix follows the format I found [here](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payments-payment-request-button-handler.php#L408), in the only other place I could find the order's payment method title being manually updated, just FYI.